profiler/profile.go

```diff
@@ -70,15 +71,15 @@ type profileType struct {
 	Delta *pprofutils.Delta
 	// Collect collects the given profile and returns the data for it. Most
 	// profiles will be in pprof format, i.e. gzip compressed proto buf data.
-	Collect func(profileType, *profiler) ([]byte, error)
+	Collect func(profileType, *profiler) (data []byte, extra []*pprofile.Profile, err error)
```

Comment:
Thanks! RE NIT 2, basically because I changed `Delta.Convert` [here](https://github.com/DataDog/dd-trace-go/blob/e38965376c99f9ce738d8ebe5a482e3db03bffc9/profiler/internal/pprofutils/delta.go#L36) to optionally have the extra profile without having to go change every caller of `Convert` in all the tests. The rest of the uses of a slice vs just one profile was just to be consistent with how I did it in `Convert` and not have to add `if extra != nil` checks.

Reply:
NIT 2: Ok, works for me 👍

Reply:
NIT 3: I agree that this change in general is clunky. I had an alternate refactoring where I pulled the delta profiling stuff into the `Collect` functions so that `Collect` would still just return the final, serialized profile. The only reason I did it this way was to keep the overall change of this PR smaller and easier to review. If you're okay with a bigger refactor, I can push up those changes to this PR?